### PR TITLE
Cleanup: avoid narrowing conversions

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: examples/halo_finder/halo_finder.cpp [hunk 1]
```diff
@@ -156,9 +156,14 @@ int main(int argc, char *argv[])
...
             auto const &halo_point = primitives(halos_indices(j));
-            halo_center[0] += halo_point[0] / halo_size;
-            halo_center[1] += halo_point[1] / halo_size;
-            halo_center[2] += halo_point[2] / halo_size;
+            // NOTE The explicit casts below are intended to silent warnings
+            // about narrowing conversion from 'int' to 'float'.  The potential
+            // issue is that 'float' can represent all integer values in the
+            // range [-2^23, 2^23] but 'int' can actually represent values in
+            // the range [-2^31, 2^31-1].
+            halo_center[0] += halo_point[0] / (float)halo_size;
```
Comment (masterleinad) [inline examples/halo_finder/halo_finder.cpp:164]: I thought we don't do C-style casts.

File: src/details/ArborX_Point.hpp [hunk 2]
```diff
@@ -37,7 +37,8 @@ class Point
 
   KOKKOS_INLINE_FUNCTION
   constexpr Point(Abomination data)
-      : Point(data.xyz[0], data.xyz[1], data.xyz[2])
+      : Point(static_cast<float>(data.xyz[0]), static_cast<float>(data.xyz[1]),
```
Comment (masterleinad) [inline src/details/ArborX_Point.hpp:40]: I feared that we would end up with this eventually. :slightly_smiling_face:

File: examples/halo_finder/halo_finder.cpp (hunk 1 above)
Comment (dalg24) [reply examples/halo_finder/halo_finder.cpp:164]: There are situations where C-style cast are OK.  If you don't like this one I can change.

**Comment by masterleinad** [comment]
Retest this please.

File: examples/halo_finder/halo_finder.cpp [hunk 3]
```diff
@@ -142,7 +142,7 @@ int main(int argc, char *argv[])
 
   if (print_sizes_centers)
   {
-    int num_halos = halos_offset.size() - 1;
+    auto const num_halos = static_cast<int>(halos_offset.size()) - 1;
```
Comment (aprokop) [inline examples/halo_finder/halo_finder.cpp:145]: Could do `auto num_halos = halos_offset.extent_int(0)-1` but whatever.

File: examples/halo_finder/halo_finder.cpp [hunk 4]
```diff
@@ -156,9 +156,14 @@ int main(int argc, char *argv[])
...
           {
             auto const &halo_point = primitives(halos_indices(j));
-            halo_center[0] += halo_point[0] / halo_size;
-            halo_center[1] += halo_point[1] / halo_size;
-            halo_center[2] += halo_point[2] / halo_size;
+            // NOTE The explicit casts below are intended to silent warnings
+            // about narrowing conversion from 'int' to 'float'.  The potential
+            // issue is that 'float' can represent all integer values in the
+            // range [-2^23, 2^23] but 'int' can actually represent values in
+            // the range [-2^31, 2^31-1].
```
Comment (aprokop) [inline examples/halo_finder/halo_finder.cpp:163]: Would link https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-narrowing-conversions.html here.

Comment (aprokop) [reply examples/halo_finder/halo_finder.cpp:163]: Maybe also tweak a comment to indicate that this is *not* about the range of representation, but about the range of integers that can be **exactly** represented by floats.

**Pull request merged**